pkg/virt-api/rest/authorizer_test.go

```diff
@@ -50,172 +52,143 @@ var _ = Describe("Authorizer", func() {
 			req.Request.Header[userHeader] = []string{"user"}
 			req.Request.Header[groupHeader] = []string{"userGroup"}
 			req.Request.Header[userExtraHeaderPrefix+"test"] = []string{"userExtraValue"}
+			req.Request.TLS = &tls.ConnectionState{}
+			req.Request.TLS.PeerCertificates = append(req.Request.TLS.PeerCertificates, &x509.Certificate{})
+
+			sarHandler = func(sar *authv1.SubjectAccessReview) (*authv1.SubjectAccessReview, error) {
+				panic("unexpected call to sarHandler")
+			}
+
+			kubeClient := fake.NewSimpleClientset()
+			kubeClient.Fake.PrependReactor("create", "subjectaccessreviews", func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+				create, ok := action.(testing.CreateAction)
+				Expect(ok).To(BeTrue())
+				sar, ok := create.GetObject().(*authv1.SubjectAccessReview)
+				Expect(ok).To(BeTrue())
+				sarOut, err := sarHandler(sar)
+				return true, sarOut, err
+			})
 
-			server = ghttp.NewServer()
-			config, err := clientcmd.BuildConfigFromFlags(server.URL(), "")
-			Expect(err).ToNot(HaveOccurred())
-
-			client, err := authorizationclient.NewForConfig(config)
-			Expect(err).ToNot(HaveOccurred())
-
-			app.client = client.SubjectAccessReviews()
-			app.userHeaders = append(app.userHeaders, userHeader)
-			app.groupHeaders = append(app.groupHeaders, groupHeader)
-			app.userExtraHeaderPrefixes = append(app.userExtraHeaderPrefixes, userExtraHeaderPrefix)
+			app = NewAuthorizorFromClient(kubeClient.AuthorizationV1().SubjectAccessReviews())
 		})
 
 		Context("Subresource api with namespaced resource", func() {
 			Context("with namespaced resource", func() {
+				sarHandlerFn := func(allowed bool) func(review *authv1.SubjectAccessReview) (*authv1.SubjectAccessReview, error) {
```

Comment:
`sarHandlerFn`, what do you say to `allowed` or `allow`? (then it would read `allowed(true)`, `allowed(false)`)